lib/utils.js

```diff
@@ -0,0 +1,131 @@
+"use strict";
+
+module.exports = (dateService, logger, URI) => {
+    /**
+     * Concats the environment and role together to make the path.
+     *
+     * @param {Object} args The user entered arguments.
+     * @return {string}
+     */
+    function buildPath(args) {
+        var path;
+
+        path = `${args.environment}/${args.role}`;
+
+        logger.debug(`Built Shelf path: ${path}`);
+
+        return path;
+    }
+
+    /**
+     * Builds a full URL based on the path provided.  The path
+     * doesn't (and should not) include the bucket name or the static
+     * "artifact" section.
+     *
+     * https://api.shelf.cwscloud.net/<bucket>/artifact/<path>
+     *
+     * You only provide the path section.
+     *
+     * @param {string} path
+     * @param {string} host
+     * @param {string} refName
+     * @return {string}
+     */
+    function buildUrl(path, host, refName) {
+        var uri;
+
+        if (path[0] === "/") {
+            // Remove first slash if one was provided.
+            path = path.slice(1);
+        }
+
+        if (path[path.length - 1] === "/") {
+            path = path.slice(0, path.length);
+        }
+
+        // TODO: Check host to see if it starts with "http://"" or "https://"
+        // TODO: Ask Kyle about why he had the dateService.now() after path
+        uri = new URI(host)
```

Comment:
It seems to me that this code might benefit from checking whether or not the host begins with `http://` or `https://`. 

Also, one might ask Kyle about why he had the dateService.now() after path.